src/frontend/src/lib/services/address.services.ts

```diff
@@ -115,12 +118,33 @@ const loadBtcAddress = async ({
 }): Promise<ResultSuccess> =>
 	loadTokenAddress<BtcAddress>({
 		tokenId,
-		getAddress: (identity: OptionIdentity) =>
-			getBtcAddress({
+		getAddress: async (identity: OptionIdentity) => {
+			const apiNetwork = mapToSignerBitcoinNetwork({ network });
+			const address = await getBtcAddress({
 				identity,
-				network: mapToSignerBitcoinNetwork({ network }),
+				network: apiNetwork,
 				nullishIdentityErrorMessage: get(i18n).auth.error.no_internet_identity
-			}),
+			});
+			try {
+				// Endpoint raises an error locally if network is not regtest.
+				if (!LOCAL || network === 'regtest') {
```

Comment:
I'm confuse why you are "loading pending transactions" within the callback that is, and should in my opinion be solely, use to load an address. Is there any particular reason?

Reply:
For different reasons, but I'm open to a refactor.

I want to load the pending transactions after I know the address.

Reply:
There is two different sign-in flows and yes have a pattern for things that need to wait depending of the feature. Let's discuss Monday offline?

Reply:
Sounds good.